webapp/components/autosize_textarea.jsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import $ from 'jquery';
+
+export default class AutosizeTextarea extends React.Component {
+    static propTypes = {
+        value: React.PropTypes.string,
+        placeholder: React.PropTypes.string,
+        onHeightChange: React.PropTypes.func
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.height = 0;
+    }
+
+    get value() {
+        return this.refs.textarea.value;
+    }
+
+    set value(value) {
+        this.refs.textarea.value = value;
+    }
+
+    componentDidUpdate() {
+        this.recalculateSize();
+    }
+
+    recalculateSize() {
+        const height = this.refs.reference.scrollHeight;
+
+        if (height > 0 && height !== this.height) {
+            const textarea = this.refs.textarea;
+            const $textarea = $(textarea);
+
+            const borderWidth = parseInt($textarea.css('border-top-width'), 10) + parseInt($textarea.css('border-bottom-width'), 10);
```

Comment:
Thanks for reminding me about that. I meant to look for a non-jquery solution, but I'm not too familiar with how [Vanilla.js](http://vanilla-js.com/) handled cascading styles.